src/index.ts

```diff
@@ -70,6 +77,15 @@ class Client {
             throw new Error('A port is required');
         }
 
+        this.isDebug = false;
+
+        this.debug = debuglog('dats', () => {
+            this.isDebug = this.debug.enabled;
+            this.debug = this.isDebug ? debug : () => undefined;
```

Comment:
If am not mistaken you are always using console.log this way. Is that what you wanted?

Reply:
Or the function passed in the options.

Reply:
Yes, I want to pass it always in debug mode.

Reply:
Doing this in this way, we reduce the overhead in production because we remove all wrapping methods.

Reply:
The argument the callback is an optimized function, without wrapping. Why not using that one? It should be very optimized.

Reply:
If a custom one is not passed as an option, of course.

Reply:
Yes, for sure if it is not passed we can use the optimized version

Reply:
Done!